trade-finance-manager-ui/server/controllers/utilisation-reports/utilisation-report-reconciliation-for-report/get-and-clear-fields-from-redirect-session-data.ts

```diff
@@ -33,21 +37,22 @@ export const getAndClearFieldsFromRedirectSessionData = (
   if (!addPaymentErrorKey) {
     return {
       errorSummary: undefined,
-      isCheckboxChecked: () => false,
+      selectedFeeRecordIds: new Set(),
     };
   }
 
-  const checkedCheckboxIds = { ...req.session.checkedCheckboxIds };
-
-  const isCheckboxChecked = (checkboxId: string): boolean => Boolean(checkedCheckboxIds[checkboxId]);
+  const checkedCheckboxIdRecords = { ...req.session.checkedCheckboxIds };
+  const checkedCheckboxIds = getPremiumPaymentsCheckboxIdsFromObjectKeys(checkedCheckboxIdRecords);
+  const selectedFeeRecordIds = getFeeRecordIdsFromPremiumPaymentsCheckboxIds(checkedCheckboxIds);
+  const selectedFeeRecordIdsSet = new Set(selectedFeeRecordIds);
 
   switch (addPaymentErrorKey) {
     case 'no-fee-records-selected':
     case 'different-fee-record-statuses':
     case 'different-fee-record-payment-currencies':
     case 'multiple-does-not-match-selected':
       clearRedirectSessionData(req);
-      return { errorSummary: getAddPaymentError(addPaymentErrorKey), isCheckboxChecked };
+      return { errorSummary: getAddPaymentError(addPaymentErrorKey), selectedFeeRecordIds: selectedFeeRecordIdsSet };
```

Comment:
Are we able to resolve the codacy reported issues here as part of this PR?

Reply:
Similar to [the other Codacy comment](https://github.com/UK-Export-Finance/dtfs2/pull/3399/files#r1733102606), not sure why this is coming up.

The argument it refers to `addPaymentErrorKey` is typed as expected (see screenshot):

![Screenshot 2024-08-29 at 09 32 18](https://github.com/user-attachments/assets/71391544-3346-4c53-9a16-5cff4060683f)

And the function sig which requires the parameter be of type `AddPaymentErrorKey`:

![Screenshot 2024-08-29 at 09 34 10](https://github.com/user-attachments/assets/b735c1a5-374f-42a3-89a2-e6df9e8785da)